### ISSUE TITLE
How to create a bidirectional many-to-many relationship?

### ISSUE THREAD

**Issue by delijah** [tommikaikkonen/redux-orm#54]
I am trying to create a bidirectional many-to-many relationship between users and groups, but it seems not to work. Here my class definitions:

```
export class Group extends ResourceModel {
    static get modelName() { return 'Group'; }

    static get fields() {
        return {
            users: many('User', 'groups'),
        };
    }

    static get defaultProps() {
        return {
            name: '',
            users: [],
        };
    }
}

export class User extends ResourceModel {
    static get modelName() { return 'User'; }

    static get fields() {
        return {
            groups: many('Group', 'users'),
        };
    }

    static get defaultProps() {
        return {
            firstName: '',
            lastName: '',
            groups: [],
        };
    }
}
```

When i create a user with groups, the state will receive a user with groups, but when i create a group with users, the state will receive a group but without users. If i remove the fields definitions on the User model, it works the opposite way. So it looks like many-to-many relationships works unidirectional only. Is this true?

Creating a user with groups (works):
```
session.Group.create({
    name: 'Administrators',
});

session.User.create({
    firstName: 'Random',
    lastName: 'Name',
    groups: [ 1 ],
});
```

Creating a group with users (doesn't work; the user and the group is created, but not the relation):
```
session.User.create({
    firstName: 'Random',
    lastName: 'Name',
});

session.Group.create({
    name: 'Administrators',
    users: [ 1 ],
});
```

**Comment by delijah** [comment]
Yes. It did not create the reverse relation.

And an open question would be: How to decide in which model to define the relation? If you have a big project with a lot of models, you just loose track on which model to define the relationship. It would be easier to define all the relationships on every model.

**Comment by forrestbice** [comment]
@delijah did you ever end up resolving this issue? I believe I'm having the same problem

**Comment by plandem** [comment]
Well, probably I see what you was talking about. Probably I got same issue in the past. Had to change way to use relations to to solve it. 

Setup:
```js
export class Group extends ResourceModel {
    static get modelName() { return 'Group'; }

    static get fields() {
        return {
            users: many('User', 'groups'),
        };
    }

    static get defaultProps() {
        return {
            name: '',
            users: [],
        };
    }
}

export class User extends ResourceModel {
    static get modelName() { return 'User'; }

    static get defaultProps() {
        return {
            firstName: '',
            lastName: '',
        };
    }
}
```

Usage:
```js

//BAD! Will have only direct relation at Group, User will not have reverse info for it.
session.User.create({
    firstName: 'Random',
    lastName: 'Name',
});

session.Group.create({
    name: 'Administrators',
    users: [ 1 ],
});

//GOOD: Both relations (direct/reverse) will be added.
var user = session.User.create({
    firstName: 'Random',
    lastName: 'Name',
});

var group = session.Group.create({
    name: 'Administrators',
});
```

And add relation info like (reverse will be auto added):
```js
group.users.add(user);
```

OR
```js
user.groups.add(group)
```

**Comment by plandem** [comment]
I asked about this in the past, but issue still exists. Not sure about opinion of author - issue or not.

**Comment by guillaumepotier** [comment]
Hi there,

Quite new here to `redux-orm` but I seem to face the same problem. How could I handle bidirectional many to many relationships between two resources?

Thanks for your help

**Comment by plandem** [comment]
reverse updated fixed at 0.9.1
